forwardbot_gitlab/models/merge_request.py

```diff
@@ -0,0 +1,214 @@
+import logging
+from typing import Any, Dict, Sequence, Tuple
+
+from odoo import api, fields, models
+from requests import Request, Session
+
+_logger = logging.getLogger(__name__)
+
+
+class MergeRequest(models.Model):
+    _name = "forwardbot_gitlab.merge.request"
+    _description = "Gitlab Merge Request (fwbot)"
+
+    name = fields.Char(help="Merge Request Title", required=True)
+    global_id = fields.Integer(
+        help="ID for this MR on the remote server. Unique across all projects"
+    )
+    project_iid = fields.Integer(help="Internal ID. Unique inside the target project.")
+    source_branch = fields.Char(required=True)
+    source_project_id = fields.Integer(required=True)
+    target_branch = fields.Char(required=True)
+    target_project_id = fields.Integer(required=True)
+    state = fields.Selection(
+        [
+            ("pending", "Pending"),
+            ("opened", "Opened"),
+            ("closed", "Closed"),
+            ("locked", "Locked"),
+            ("merged", "Merged"),
+        ],
+        default="pending",
+        required=True,
+    )
+    forward_port_id = fields.Many2one("forwardbot_gitlab.merge.request")
+    processed = fields.Boolean(default=False)
+
+    _sql_constraints = [
+        (
+            "unique_global_id",
+            "UNIQUE (global_id)",
+            "Two MRs can't have the same ID",
+        ),
+    ]
+
+    required_merge_request_fields = [
+        "id",
+        "iid",
+        "title",
+        "target_branch",
+        "source_branch",
+        "state",
+    ]
+
+    @api.multi
+    def gen_api_merge_request(self, base_url: str, description: str = "") -> Request:
+        self.ensure_one()
+        req = Request(
+            "POST",
+            f"{base_url}/api/v4/projects/{self.target_project_id}/merge_requests",
+            json={
+                "id": self.source_project_id,
+                "source_branch": self.source_branch,
+                "target_branch": self.target_branch,
+                "title": self.name,
+                "description": description,
+            },
+            headers={"Content-Type": "application/json"},
+        )
+
+        return req
+
+    @api.multi
+    def create_forward_port(self) -> "MergeRequest":
+        self.ensure_one()
+        if self.forward_port_id:
+            return self.forward_port_id
+
+        try:
+            forward_branch = str(float(self.target_branch) + 1)
```

Comment:
It is good for me and It will work for 99% of the cases

But we had one time in the past a branch called `saas-x` (not float format)

We could think in the future how to support not-float-format branches

Something like:
`runbot_branch.foward_branch_id = fields.Many2one("runbot.branch")` and maybe auto-choose using this way (float-format) but we can assign another one or skip a middle version of something else.

But for now is good, only add a comment TODO related, please

Reply:
What happens if we're forward-porting from 14.0 to 16.0?

14.0 + 1 = 15.0.

Please take into account skipping versions is a very common case, most customers don't migrate to the very next version.

Reply:
> What happens if we're forward-porting from 14.0 to 16.0?
> 
> 14.0 + 1 = 15.0.
> 
> Please take into account skipping versions is a very common case, most customers don't migrate to the very next version.

Good point. I tried to keep this as simple as possible but I think the bot will also need to keep track of the repo branches. Keeping track of this will also stop the bot from making MRs for something like 17.0. In this case the flow will be changed to make a forward port to the next branch (that exists) available

Reply:
Maybe next sticky branch

But How do you know what is next if you have saas-14 and master

Maybe, the fw_branch_id field could help us a lot here

FYI we have sticky field in the branches to detect the stable ones

Reply:
For now, I agree on converting branches to float, I just don't agree on adding 1. I think we would need to:
- Convert all branches to float
- Sort them and save the result to a list
- Get index of current branch on the sorted list and add 1 to that index, that should return the next version

How does Odoo this same thing?

Maybe a config param

Reply:
That could work to auto-assign the fw_branch_id with a value by default

But even we are able to change it from the runbot.branch itself

Consider master, main, saas-* doesn't complain this standard